### PR TITLE
Assert DF equality in P2P tests

### PR DESCRIPTION
**Pull request by hendrikmakait**
Asserts that P2P shuffling return the correct results

- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

File: distributed/shuffle/tests/test_shuffle.py [hunk 1]
```diff
@@ -144,12 +144,12 @@ async def test_basic_cudf_support(c, s, a, b):
...
-    assert out.npartitions == df.npartitions
-    x, y = c.compute([df.x.size, out.x.size])
-    x = await x
-    y = await y
-    assert x == y
+    shuffled = dd.shuffle.shuffle(df, "x", shuffle="p2p")
+    assert shuffled.npartitions == df.npartitions
+
+    result, expected = c.compute([shuffled, df])
+    result, expected = await c.gather([result, expected])
```
Comment (wence-) [inline distributed/shuffle/tests/test_shuffle.py:151]: First of these lines is redundant without checking for equality, no?

or actually no, I am confused compute brings the "complete" thing to the host, what does the gather then do?

**Review by wence- — COMMENTED** [review]
Think I am confused about what the `gather` of the `compute`d results is supposed to do.

File: distributed/shuffle/tests/test_shuffle.py (hunk 1 above)
Comment (hendrikmakait) [reply distributed/shuffle/tests/test_shuffle.py:151]: These tests are run in async mode, so we still have to await the results.

Comment (wence-) [reply distributed/shuffle/tests/test_shuffle.py:151]: Ah, maybe `c.gather([shuffled, df], sync=True)` or `result, expected = await c.gather(c.compute([shuffled, df]))`?

**Comment by wence-** [comment]
Not sure the current state of tests, but a variety of fails (including a segfault on `ubuntu-latests 3.9, queue, ci1`), not all these failures exhibit in every failing run:
```
distributed/tests/test_utils_test.py::test_popen_timeout
FAILED distributed/tests/test_stress.py::test_chaos_rechunk - asyncio.exceptions.TimeoutError
FAILED distributed/shuffle/tests/test_merge.py::test_basic_merge[none-inner] - UserWarning: Timed out waiting for sampling thread.
FAILED distributed/tests/test_spans.py::test_before_first_task_finished - AssertionError: assert 1692797492.5552566 < 1692797492.5537772
FAILED distributed/tests/test_stress.py::test_chaos_rechunk - asyncio.exceptions.TimeoutError
FAILED distributed/tests/test_worker.py::test_lifetime - TimeoutError

Fatal Python error: Aborted

Thread 0x00007ff037872740 (most recent call first):
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/psutil/_common.py", line 765 in open_binary
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/psutil/_pslinux.py", line 1885 in memory_info
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/psutil/_pslinux.py", line 1653 in wrapper
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/psutil/__init__.py", line 1063 in memory_info
  File "/usr/share/miniconda3/envs/dask-distributed/lib/python3.9/site-packages/psutil/_common.py", line 478 in wrapper
  File "/home/runner/work/distributed/distributed/distributed/utils_perf.py", line 181 in _gc_callback
/home/runner/work/_temp/9cdaa655-4ad4-41ce-8bab-973d29911a9e.sh: line 10:  2386 Aborted                 (core dumped) p
```

**Comment by hendrikmakait** [comment]
> Not sure the current state of tests, but a variety of fails (including a segfault on `ubuntu-latests 3.9, queue, ci1`), not all these failures exhibit in every failing run:

This looks - unfortunately - like the normal amount of noise we currently experience on CI. :( Test failures should be unrelated from what I can tell.

**Review by wence- — APPROVED** [review]
Thanks

**Pull request merged**